3_cluster/src/gage_year_coverage_by_cluster.R

```diff
@@ -0,0 +1,161 @@
+  
+  p1<-ggplot(df, aes(x= waterYear, y = ID, fill = n)) + geom_tile() +
+    scale_fill_viridis_c(option = "plasma")+
+    theme(axis.text.y = element_blank(),
+          axis.ticks.y = element_blank())+
+    scale_x_continuous(breaks = seq(1900, 2020, 20))+
+    facet_grid(cluster~., scales="free", space="free")+
+    ggtitle(paste("data coverage for " , cluster_name))
+  if (estimated_data == FALSE) {
+    fileout<- file.path(dir_out, paste0(plot_col, "quants_data_coverage-no_estimated_data.png"))
+  } else {
+    fileout<- file.path(dir_out, paste0(plot_col, "quants_data_coverage- with_estimated_data.png"))
+  }
+ 
+  ggsave(filename = fileout,
+         plot = p1,
+         device = png)
+  
+  
+  return(fileout)
+  
+}
+
+
+estimated_data_quantiles <- function(screened_site_list, cluster_table,dv_data_dir,dir_out){
+  #' @description Function to determine and plot the quantiles of the estimated data by cluster
+  #' 
+  #' @param screened_site_list p1_screened_site_list target. 
+  #' @param cluster_table data frame with ID (gage ID), and any number of columns containing
+  #' cluster id for different groups of quantiles
+  #' @param dv_data_dir directory where daily NWIS streamflow data were saved (currently ./1_fetch/out")
+  #' @param dir_out directory to save plots
+  
+  
+  gage_list<- unique(cluster_table$ID) 
+  file_list <- paste0(dv_data_dir, "/", gage_list, "_dv.csv") 
+  #removes the data files for the few gages that have been combined
+  file_list <- file_list[which(file.exists(file_list))]
+  estimated_data_allgages <- purrr::map_df(file_list, get_estimated_data)
+  
+  df_plot<-estimated_data_allgages %>%
+    left_join(.,cluster_table, by="ID")%>%
+    rename(midhigh_cluster = midhigh, high_cluster = high)
+  
+  
+  p1 <- ggplot(df_plot, aes(quantile)) + geom_histogram() + 
+    facet_wrap(~midhigh_cluster, scales= "free") +
+    ggtitle("Quantiles of estimated data for midhigh clusters")
+  
+  p2 <- ggplot(df_plot, aes(quantile)) + geom_histogram() + 
+    facet_wrap(~high_cluster, scales= "free") +
+    ggtitle("Quantiles of estimated data for high clusters")
+  
+  
+  fname <- paste0("Estimated_data_quantiles.png")
+  fileout <- file.path(dir_out, fname)  
+  ggsave(filename = fileout, 
+         plot =   plot_grid(p1,p2, ncol = 1, label_size = 8))
+  
+  return(fileout)
+  
+}
+
+
+
+get_estimated_data<- function(fname){
+  #' @description helper function for estimated_data_quantile to read in daily data for a single file
+  #' @param fname is the file path to the dialy csv file
+  
+  station.ID <- substr(fname, start = 15, stop = 22)
+  df<- read_csv(fname, col_types = cols()) 
+  if("discharge" %in% names(df)){
+    df <- df %>% 
+      mutate(ID = station.ID) %>%
+      mutate(quantile = ecdf(discharge)(discharge))
+    
+    df_estimated  <- df %>%
+      filter(., grepl("e", discharge_cd)) %>%
+      select(ID, Date, discharge, discharge_cd, quantile)
+    return(df_estimated)   
+    
+  } #end if
+  
```

Comment:
Do you want this to return an empty df when 'discharge' is not in names(df)?

Reply:
This was in there because there are 4 files which have more than one discharge column in NWIS for some reason.  Instead of trying to figure it out what to do with them for this pr, I just excluded them.  I figured 4 sites wouldn't make a big difference in the plots.  But - if we want these plots for a paper, and we want to be more precise, I can go back and see how we handled it in the data screening section and get them in there.